### ISSUE TITLE
Refactoring: New Screenwrapper Implementation

### ISSUE THREAD

**Issue by antfarmar** [antfarmar/Unity-3D-Asteroids#31]
I have implemented a more stable and robust screen wrapper, found in commit  48b988d.

I pretty much wrote it quickly to get the ideas out of my head and onto the screen.
It therefore has minimal regards for readability, although not entirely. So far it is 1 large `Update` method.

It works well :+1:, but readability & flow could be improved. Some duplicated code may exist as well.

So I will be improving on it's current state as time permits, mostly by refactoring it to make it read like prose.

**Comment by antfarmar** [comment]
A preliminary refactoring: commit 9fe5b93
[`Screenwrapper.cs`](https://github.com/antfarmar/Unity-3D-Asteroids/blob/9fe5b93ca6e621319cff36cca6c21b8527eddca6/Assets/Asteroids/Scripts/MicroBehaviours/ScreenWrapper.cs)

**Comment by antfarmar** [comment]
My general attitude towards writing readable code is changing: :+1: :grinning:  
- I used to avoid writing verbose variable/method names.
- I kept variable/method usage to an absolute minimum.
- I used to prefer clever hacks over clear code.
- I kept refactoring iterations to a minimum.

These bad habits may have been a by-product of casually participating in timed coding competitions (TopCoder, HackerRank) and coding challenges (Project Euler, Rosalind Project), where solutions are written hastily or just simply solve one-off problems and don't require maintenance or revisits.

Large software projects with several team members is a different discipline entirely, and should be approached as such.
